### PR TITLE
Support ghc-8.10 and ghc-9.2

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,28 +12,27 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        cabal: ["3.2"]
-        ghc: ["8.6.5", "8.8.4", "8.10.2"]
+        ghc: ["8.10.7", "9.2.2"]
```
Comment (angerman) [inline .github/workflows/ci.yml:15]: This should be 9.2.4?

File: src/NoThunks/Class.hs [hunk 2]
```diff
@@ -58,7 +59,9 @@ import Data.Time
 import Data.Void (Void)
 import Data.Word
 import GHC.Stack
+#if !MIN_VERSION_base(4,15,0)
```
Comment (angerman) [inline src/NoThunks/Class.hs:62]: Can/Should we add a comment what happened in 4.15.0+?

**Review by angerman — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (erikd) [reply .github/workflows/ci.yml:15]: yes, will fix that

File: src/NoThunks/Class.hs (hunk 2 above)
Comment (erikd) [reply src/NoThunks/Class.hs:62]: I am not even sure why that was needed.

Comment (coot) [reply src/NoThunks/Class.hs:62]: @angerman in `4.16.0+` `Natural` is exported from `GHC.TypeLits` which is imported a few lines above this place.  It's possible that the code does not compile with `ghc-9.0`.

Comment (coot) [reply src/NoThunks/Class.hs:62]: Indeed it doesn't:
```

src/NoThunks/Class.hs:457:28: error:
    Not in scope: type constructor or class ‘Natural’
    |
457 | deriving via OnlyCheckWhnf Natural instance NoThunks Natural
    |                            ^^^^^^^

src/NoThunks/Class.hs:457:54: error:
    Not in scope: type constructor or class ‘Natural’
    |
457 | deriving via OnlyCheckWhnf Natural instance NoThunks Natural
    |                                                      ^^^^^^^
```

**Pull request merged**